pycrostates/cluster/aahc.py

```diff
@@ -0,0 +1,408 @@
+"""Class and functions to use Atomize and Agglomerate Hierarchical Clustering
+   (AAHC)."""
+
+from functools import wraps as func_wraps
+from pathlib import Path
+from typing import Any, Optional, Tuple, Union
+
+import numpy as np
+from mne import BaseEpochs
+from mne.io import BaseRaw
+from numpy.typing import NDArray
+
+from .._typing import Picks
+from ..utils import _corr_vectors
+from ..utils._checks import _check_type
+from ..utils._docs import copy_doc, fill_doc
+from ..utils._logs import _set_verbose, logger
+from ._base import _BaseCluster
+
+# if we have numba, use its jit interface
+try:
+    from numba import njit
+except ImportError:
+    def njit(cache=False):
+        def decorator(func):
+            @func_wraps(func)
+            def wrapper(*args, **kwargs):
+                return func(*args, **kwargs)
+
+            return wrapper
+
+        return decorator
+
+
+@fill_doc
+class AAHCluster(_BaseCluster):
+    r"""Atomize and Agglomerate Hierarchical Clustering (AAHC) algorithm.
+
+    See :footcite:t:`Murray2008` for additional information.
+
+    Parameters
+    ----------
+    %(n_clusters)s
+    ignore_polarity : bool
+        If true, polarity is ignored when computing distances.
+    normalize_input : bool
+        If set, the input data is normalized along the channel dimension.
+    tol : float
+        Relative tolerance with regards estimate residual noise in the cluster
+        centers of two consecutive iterations to declare convergence.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+
+    def __init__(
+        self,
+        n_clusters: int,
+        ignore_polarity: bool = True,
+        normalize_input: bool = False,
+        tol: float = 1e-6,
+    ):
+        super().__init__()
+
+        self._n_clusters = _BaseCluster._check_n_clusters(n_clusters)
+        self._cluster_names = [str(k) for k in range(self.n_clusters)]
+
+        self._ignore_polarity = AAHCluster._check_ignore_polarity(
+            ignore_polarity
+        )
+        self._normalize_input = AAHCluster._check_ignore_polarity(
+            normalize_input
+        )
+        self._tol = AAHCluster._check_tol(tol)
+
+        # fit variables
+        self._GEV_ = None
+
+    def _repr_html_(self, caption=None):
+        from ..html_templates import repr_templates_env
+
+        template = repr_templates_env.get_template("AAHCluster.html.jinja")
+        if self.fitted:
+            n_samples = self._fitted_data.shape[-1]
+            ch_types, ch_counts = np.unique(
+                self.get_channel_types(), return_counts=True
+            )
+            ch_repr = [
+                f"{ch_count} {ch_type.upper()}"
+                for ch_type, ch_count in zip(ch_types, ch_counts)
+            ]
+            GEV = int(self._GEV_ * 100)
```

Comment:
I feel like the convertion to interger could lead to missunderstanding ( GEV = 50.56% but displays 50%).
Maybe something like `GEV = f'{self._GEV_ * 100:.2f}'` would be more suitable.

I see that `ModKMeans `has the same issue. Feel free to correct it for both Classes, otherwise I'll do it in another PR.

Reply:
Resolved in https://github.com/vferat/pycrostates/pull/92/commits/1161e8eeba3a0c522d5e5577454d45423393527a